src/home_assistant.rs

```diff
@@ -203,6 +270,51 @@ impl Config {
         }))
     }
 
+    fn number_percent(
+        &self,
+        name: &str,
+        label: &str,
+        holding_register: Register,
+    ) -> Result<Option<mqtt::Message>> {
+        if !enabled(self.mqtt_config.homeassistant().controls(), name) {
+            return Ok(None);
+        }
+
+        let config = Number {
+            name: label.to_string(),
+            state_topic: format!(
+                "{}/{}/hold/{:?}",
+                self.mqtt_config.namespace(),
+                self.inverter.datalog(),
+                holding_register,
```

Comment:
This (and the set) are producing content like this:

```json
"state_topic":"lxp/xxx/hold/AcChargeEndSocLimit",
"command_topic":"lxp/cmd/xxx/set/hold/AcChargeEndSocLimit"
```

Which is slightly bizarre and unexpected. I think we might need to cast here, so probably `holding_register as i16`. I thought the `#[repr(i16)]` on the enum definition might do that automatically but maybe I'm expecting too much :)